pkg/model/filters/filters.go

```diff
@@ -52,3 +55,30 @@ func Parse(raw string) (MultiQueries, error) {
 	}
 	return parsed, nil
 }
+
+func SplitForReportersMerge(q SingleQuery) (SingleQuery, SingleQuery) {
+	// If FlowDirection is enforced, skip merging both reporters
+	for _, m := range q {
+		if m.Key == fields.FlowDirection {
+			return q, nil
+		}
+	}
+	// The rationale here is that most traffic is duplicated from ingress and egress PoV, except cluster-external traffic.
+	// TODO: this assumption seems wrong wrt Dropped traffic; need to figure out how to deal with it.
+	// Merging is done by running a first query with FlowDirection=INGRESS and another with FlowDirection=EGRESS AND SrcOwnerName is empty,
+	// which stands for cluster-external.
+	// (Note that we use SrcOwnerName both as an optimization as it's a Loki index,
+	// and as convenience because looking for empty fields won't work if they aren't indexed)
```

Comment:
hey, @jpinsonneau @msherif1234 @OlivierCazade , would like some advice here.
I think the assumption I made for merged reporters won't work for dropped traffic: I guess there is no such duplication on ingress and egress there? If confirmed, we should do specific queries for drop traffic I guess